File: src/ruby/gem/appcelerator/templates/test_service.rb

class TestService < Appcelerator::Service
  
  Service 'app.test.message.request', :testMessage, 'app.test.message.response'
  
  def testMessage
      p "received message: #{params.inspect}"
      msg = params["message"]
      {"message"=>"I received from you: #{msg}","success"=>true}
  end
  
end

File: src/dotnet/ServiceBroker/TestService.cs
using System;
using System.Collections.Generic;
using System.Text;
using Appcelerator;
using NetServ.Net.Json;
namespace HelloWorld
{
    public class TestService
    {
        [Service("r:hello.request", "r:hello.response")]
        public void HelloWorld(Message request, ref Message response)
        {
            JsonObject data = request.Data;
            data.Add("Hello","world");
            response.Data = data;
        }
    }
}
